src/modules/mahjong/PointValidator/PointValidator.ts

```diff
@@ -0,0 +1,338 @@
+      },
+      allHands: [],
+    };
+
+    let largestHand: HandPointResults = {
+      melds: [],
+      points: 0,
+      hands: [],
+      tiles: [],
+    };
+
+    if (handStructureResults.isThirteenTerminals) {
+      largestHand.points = HKHandMapper.THIRTEEN_ORPHANS.points;
+      largestHand.tiles = handStructureResults.valid[0].originalTiles;
+
+      result.largestHand = largestHand;
+      result.allHands.push(largestHand);
+      return result;
+    }
+
+    const { valid } = handStructureResults;
+
+    valid.forEach((vp) => {
+      let points = 0;
+      const hands: HandDefinition[] = [];
+
+      const consecutive: HandDefinition = PointValidator.validateAllConsecutives(vp);
+      let triplets = {
+        points: 0,
+        name: '',
+      }; // placeholder
+      let kongs = {
+        points: 0,
+        name: '',
+      };
+
+      // Check first if the hand is allConsecutives or allTriplets as these are common hands
+      if (consecutive.points > 0) {
+        points += consecutive.points;
+        hands.push(consecutive);
+      } else {
+        // Need to check Kong before Triplets as validate all triplets will yield correct for all Kong
+        kongs = PointValidator.validateAllKongs(vp);
+        if (kongs.points > 0) {
+          points += kongs.points;
+          hands.push(kongs);
+        } else {
+          triplets = PointValidator.validateAllTriplets(vp);
+          if (triplets.points > 0) {
+            points += triplets.points;
+            hands.push(triplets);
+          }
+        }
+      }
+
+      if (consecutive.points === 0) {
+        const handsToCheck = [
+          PointValidator.validateAllHonors,
+          PointValidator.validateSmallAndLargeDragons,
+          PointValidator.validateSmallAndLargeWinds,
+        ];
+
+        handsToCheck.forEach((fn) => {
+          const resultOfFn = fn(vp);
+          if (resultOfFn.points > 0) {
+            points += resultOfFn.points;
+            hands.push(resultOfFn);
+          }
+        });
+      }
+
+      // always check for purity/semi-pure as they can stack with allConsecutives/allTriplets or be standalone
+      const purity = PointValidator.validatePurity(vp);
+      if (purity.points > 0) {
+        points += purity.points;
+        hands.push(purity);
+      }
+
+      if (purity.points === 0) {
+        const semiPure = PointValidator.validateSemiPurity(vp);
+        if (semiPure.points > 0) {
+          points += semiPure.points;
+          hands.push(semiPure);
+        }
+      }
+
+      const resultOfThisVP = {
+        points,
+        melds: vp.melds,
+        hands,
+        tiles: vp.originalTiles,
+      };
+
+      if (resultOfThisVP.points > largestHand.points) largestHand = resultOfThisVP;
+      result.allHands.push(resultOfThisVP);
+    });
+
+    result.largestHand = largestHand;
+    if (result.largestHand.points > PointValidator.MAX_POINTS) result.largestHand.points = PointValidator.MAX_POINTS;
+    return result;
```

Comment:
Should there be a statement to round to 0 if the result points is less than 3

Reply:
I don't think so. 2 Points is still 2 points. We can just have a later check that checked if the points calculated is greater or equal to 3.